File: lvl1 - Team Membership Tasks/Task5-Matrix/Matrix.py
class Matrix:
    def __init__(self, a, b):
        self.Row = a
        self.Column = b
        self.mat = [[0 for i in range(self.Column)] for j in range(self.Row)]
    def init(self):
        for i in range(self.Row):
            for j in range(self.Column):
                number = float(input(f"Matrix [{i}][{j}] : "))
                self.mat[i][j] = number
        self.print()
    def reverser3(self):
        determinant = 0
        for i in range(3):
            determinant += (self.mat[0][i] * (self.mat[1][(i+1)%3] * self.mat[2][(i+2)%3] - self.mat[1][(i+2)%3] * self.mat[2][(i+1)%3]))
        print(f"\ndeterminant: {determinant}\n")
        for i in range(3):
            for j in range(3):
                print(((self.mat[(j+1)%3][(i+1)%3] * self.mat[(j+2)%3][(i+2)%3]) - (self.mat[(j+1)%3][(i+2)%3] * self.mat[(j+2)%3][(i+1)%3]))/ determinant, end="\t")
            print("\n")
    def reverser2(self):
        Test3 = [[0 for i in range(2)] for j in range(2)]
        determinant = ((self.mat[0][0] * self.mat[1][1]) - (self.mat[0][1] * self.mat[1][0]))
        for i in range(self.Row):
            for j in range(self.Column):
                Test3[i][j] = ((self.mat[(j+1)%2][(i+1)%2] * self.mat[(j+2)%2][(i+2)%2]) - (self.mat[(j+1)%2][(i+2)%2] * self.mat[(j+2)%2][(i+1)%2]))/ determinant
        for i in range(self.Row):
            for j in range(self.Column):
                print(f" {Test3[i][j]} ", end="")
            print("\n")
    def Identity(self, num):
        for row in range(num):
            for col in range(num):
                if row == col:
                    print("1", end=" ")
                else:
                    print("0", end=" ")
            print("\n")
    def __add__(self, x):
        Test3 = Matrix(self.Row, self.Column)
        for i in range(self.Row):
            for j in range(self.Column):
                Test3.mat[i][j] = self.mat[i][j] + x.mat[i][j]
        Test3.print()
    def __sub__(self, x):
        Test3 = Matrix(self.Row, self.Column)
        for i in range(self.Row):
            for j in range(self.Column):
                Test3.mat[i][j] = self.mat[i][j] - x.mat[i][j]
        Test3.print()
    def __mul__(self, x):
        Test3 = Matrix(self.Row, self.Column)
        for i in range(self.Row):
            for j in range(self.Column):
                Test3.mat[i][j] = 0
                for k in range(self.Column):
                    Test3.mat[i][j] += self.mat[i][k] * (x.mat[k][j])
        Test3.print()
    def print(self):
        for i in range(self.Row):
            for j in range(self.Column):
                print(f" {self.mat[i][j]} ", end="")
            print("\n")
    def __del__(self):
        pass

while True:
    number = int(input("enter the number row and column\n"))
    Test = Matrix(number, number)
    Test.init()
    Test1 = Matrix(number, number)
    Test1.init()
    print("Addition of two given Matrices is : ")
    Test + Test1
    print("Subtraction of two given Matrices is : ")
    Test - Test1
    print("Multiplication of two given Matrices is : ")
    Test * Test1
    print("\nReversed Matrix 1\n")
    Test.reverser3()
    print("\nReversed Matrix 2\n")
    Test1.reverser2()
    c = input("\nDo you want to continue? (y/n)\n")
    if c.lower() == "n":
        break

File: lvl1 - Team Membership Tasks/Task5-Matrix/Matrix.cpp
#include <iostream>
using namespace std;

class Matrix
{
private:
	int Row;
	int Column;
	float** mat;
public:
	int temp;
	int Identity(int num);
	void init();
	void print();
	void reverser3();
	void reverser2();
	void operator+(Matrix x);
	void operator-(Matrix x);
	void operator*(Matrix x);
	Matrix(int a, int b);
	Matrix();
	~Matrix();

};
///////////////////////////////////////////////////////////
Matrix::Matrix(int a, int b)
{
	Row = a;
	Column = b;
	mat = new float* [Row];
	for (int i = 0; i < Column; ++i)
		mat[i] = new float[Column];
}
//////////////////////////////////////////////////////////
void Matrix::init()
{
	float number;
	for (int i = 0; i < Row; i++) {
		for (int j = 0; j < Column; j++) {
			cout << "Matrix " << "[" << i << "]" << "[" << j << "] : ";
			cin >> number;
			mat[i][j] = number;
		}

	}
	print();
}
//////////////////////////////////////////////////////////
void Matrix::reverser3()
{
   int  i, j;
   float determinant = 0;
   //finding determinant
    for(i = 0; i < 3; i++)
       determinant =+ (mat[0][i] * (mat[1][(i+1)%3] * mat[2][(i+2)%3] - mat[1][(i+2)%3] * mat[2][(i+1)%3]));
       cout<<"\n\ndeterminant: "<<determinant<<endl;
    for(i = 0; i < 3; i++){
      for(j = 0; j < 3; j++)
        cout<<((mat[(j+1)%3][(i+1)%3] * mat[(j+2)%3][(i+2)%3]) - (mat[(j+1)%3][(i+2)%3] * mat[(j+2)%3][(i+1)%3]))/ determinant<<"\t";
            cout<<"\n";
    }    
}
//////////////////////////////////////////////////////////
void  Matrix::reverser2() {
	float Test3[2][2];
	float determinant;
	Test3[0][0] = mat[1][1];
	Test3[0][1] = -mat[0][1];
	Test3[1][0] = -mat[1][0];
	Test3[1][1] = mat[0][0];

	determinant = ((mat[0][0] * mat[1][1]) - (mat[0][1] * mat[1][0]));

	for (int i = 0; i < Row; i++)
		for (int j = 0; j < Column; j++) {
			Test3[i][j] *= (1 / determinant);
		}

	for (int i = 0; i < Row; i++) {
		for (int j = 0; j < Column; j++)
			cout << " " << Test3[i][j] << " ";
		cout << endl;
	}
}

//////////////////////////////////////////////////////////
int Matrix::Identity(int num)
{
	int row, col;

	for (row = 0; row < num; row++)
	{
		for (col = 0; col < num; col++)
		{

			if (row == col)
				cout << 1 << " ";
			else
				cout << 0 << " ";
		}
		cout << endl;
	}
	return 0;
}

//////////////////////////////////////////////////////////
void Matrix::operator+(Matrix x)
{
	Matrix Test3(Row, Column);


	for (int i = 0; i < Row; i++) {
		for (int j = 0; j < Column; j++) {


			Test3.mat[i][j] = mat[i][j]
				+ x.mat[i][j];
		}
	}

	Test3.print();
}
//////////////////////////////////////////////////////////
void Matrix::operator-(Matrix x)
{
	Matrix Test3(Row, Column);

	for (int i = 0; i < Row; i++) {

		for (int j = 0; j < Column; j++) {

			Test3.mat[i][j] = mat[i][j]
				- x.mat[i][j];
		}
	}

	Test3.print();
}
//////////////////////////////////////////////////////////
void Matrix::operator*(Matrix x)
{
	Matrix Test3(Row, Column);




	for (int i = 0; i < Row; i++) {

		for (int j = 0; j < Column; j++) {

			Test3.mat[i][j] = 0;

			for (int k = 0; k < Column; k++) {
				Test3.mat[i][j] += mat[i][k]
					* (x.mat[k][j]);
			}
		}
	}
	Test3.print();

}
//////////////////////////////////////////////////////////
void Matrix::print() {

	for (int i = 0; i < Row; i++) {
		for (int j = 0; j < Column; j++)
			cout << " " << mat[i][j] << " ";
		cout << endl;
	}
}
//////////////////////////////////////////////////////////
Matrix::~Matrix()
{
}
//////////////////////////////////////////////////////////
int main()
{
    int number;
    char c;
    cout<<"enter the number row and column \n";
    cin>> number;
    while (true)
    {
        
                Matrix Test(number,number);
	            Test.init();
                Matrix Test1(number, number);
	            Test1.init();
                cout << "Addition of two given"
		        << " Matrices is : \n";
                Test + Test1;
                cout << "Subtraction of two given"
		        << " Matrices is : \n";
	            Test - Test1;
            	cout << "Multiplication of two"
		        << " given Matrices is : \n";
	            Test* Test1;
                cout << "\n\nReversed Matrix 1\n\n";
                Test.reverser3();
                cout << "\n\nReversed Matrix 2\n\n";
                Test1.reverser2();
                cout<<"\n do you want continue\n";
                cin>>c;
                if (c=='n'|| c=='N')
                 break;
            
    }
    



}
  // Persian Language (Farsi)

	// // ماتریس کهاد :
	// matrix3[0][0] = (matrix[1][1]*matrix[2][2])-(matrix[2][1]*matrix[1][2]);
	// matrix3[1][0] = (matrix[0][1]*matrix[2][2])-(matrix[2][1]*matrix[0][2]);
	// matrix3[2][0] = (matrix[0][1]*matrix[1][2])-(matrix[1][1]*matrix[0][2]);
	// matrix3[0][1] = (matrix[1][0]*matrix[2][2])-(matrix[2][0]*matrix[1][2]);
	// matrix3[1][1] = (matrix[0][0]*matrix[2][2])-(matrix[2][0]*matrix[0][2]);
	// matrix3[2][1] = (matrix[0][0]*matrix[1][2])-(matrix[1][0]*matrix[0][2]);
	// matrix3[0][2] = (matrix[1][0]*matrix[2][1])-(matrix[2][0]*matrix[1][1]);
	// matrix3[1][2] = (matrix[0][0]*matrix[2][1])-(matrix[2][0]*matrix[0][1]);
	// matrix3[2][2] = (matrix[0][0]*matrix[1][1])-(matrix[1][0]*matrix[0][1]);
	// //ماتریس همساز :
	// matrix3[0][1] *= -1;
	// matrix3[1][0] *= -1;
	// matrix3[1][2] *= -1;
	// matrix3[2][1] *= -1;
	// // ماتریس الحاقی :
	// tmp = matrix3[0][1];
	// matrix3[0][1] = matrix3[1][0];
	// matrix3[1][0] = tmp;
	
	// tmp = matrix3[0][2];
	// matrix3[0][2] = matrix3[2][0];
	// matrix3[2][0] = tmp;
	
	// tmp = matrix3[1][2];
	// matrix3[1][2] = matrix3[2][1];
	// matrix3[2][1] = tmp;
	// //  دترمینان :
	// determinant = (
	// (matrix[0][0]*(matrix[1][1]*matrix1[2][2] - matrix[2][1]*matrix[1][2]))-
	// (matrix[0][1]*(matrix[1][0]*matrix1[2][2] - matrix[1][2]*matrix[2][0]))+
	// (matrix[0][2]*(matrix[1][0]*matrix1[2][1] - matrix[2][0]*matrix[1][1]))
	// );
	// // حاصل‌ضرب ماتریس الحاقی در معکوس دترمینان :
	// for(int i=0;i<row;i++)
	// 	for(int j=0;j<column;j++){
	// 		matrix3[i][j] *= (1/determinant);
	// 	}